forge-game/src/main/java/forge/game/ability/effects/RingTemptsYouEffect.java

```diff
@@ -0,0 +1,120 @@
+package forge.game.ability.effects;
+
+import forge.ImageKeys;
+import forge.card.CardRarity;
+import forge.game.Game;
+import forge.game.ability.AbilityFactory;
+import forge.game.card.Card;
+import forge.game.card.CardCollection;
+import forge.game.player.Player;
+import forge.game.spellability.AbilitySub;
+import forge.game.spellability.SpellAbility;
+import forge.game.trigger.Trigger;
+import forge.game.trigger.TriggerHandler;
+import forge.game.trigger.TriggerType;
+import forge.game.zone.ZoneType;
+
+public class RingTemptsYouEffect extends EffectEffect {
+
+    @Override
+    protected String getStackDescription(SpellAbility sa) {
+        final StringBuilder sb = new StringBuilder();
+        sb.append("The Ring tempts " + sa.getActivatingPlayer());
+        return sb.toString();
+    }
+
+    @Override
+    public void resolve(SpellAbility sa) {
+        Player p = sa.getActivatingPlayer();
+        Game game = p.getGame();
+        Card card = sa.getHostCard();
+        Card theRing = null;
+        for (Card c : p.getCardsIn(ZoneType.Command)) {
+            if (c.getName().equals("The Ring")) {
+                theRing = c;
+                break;
+            }
+        }
+
+        int level = 0;
+        if (theRing == null) {
+            // Create the ring with first level
+            String image = ImageKeys.getTokenKey("the_ring");
+            theRing = createEffect(sa, p, "The Ring", image);
+            theRing.setSetCode(card.getSetCode());
+            theRing.setRarity(CardRarity.Common);
+
+            game.getTriggerHandler().suppressMode(TriggerType.ChangesZone);
+            theRing = p.getGame().getAction().moveTo(p.getZone(ZoneType.Command), theRing, sa);
+            game.getTriggerHandler().clearSuppression(TriggerType.ChangesZone);
+        } else {
+            level = Integer.parseInt(theRing.getSVar("RingLevel"));
+        }
+
+        level += 1;
+        theRing.setSVar("RingLevel", String.valueOf(Math.min(level, 4)));
```

Comment:
just cap the level at 4 there so it can't even go over level 4 anyway?

no need to check later?